### ISSUE TITLE
Cannot train 'simple' neural net

### ISSUE THREAD

**Issue by KarthikGanesan88** [tiny-dnn/tiny-dnn#318]
Hi, I trained and tried out the example MNIST program and it seems to be working fine. This network consists of Conv -> pool -> conv -> (connection table) -> pool -> conv -> FC. 

However, I am trying to implement the much simpler network given in the example of FC(32*32,128) -> FC(128,10). But I'm not able to train this network successfully! 

For the first network, after training for 1 epoch i get 9636/10000 correct. But for the 2nd network, i am getting 958/10000! Also the training seems to go on forever and doesnt stop after 6 epochs and each epoch always gets the same 958/10000! I tried using make_mlp and that bumps me up slightly to 1135/10000. 

I tried another example from the Tiny-dnn howto i.e Conv -> Ave -> FC<tan_h> -> FC<softmax>. And that one seems to train ok. So it just seems to be a problem training 'purely' fully connected networks!

Does anyone know what might be going on here? I understand that a simple FC only network wont do as well as the complicated first network I have but this seems a little too bad!

**Comment by nyanp** [comment]
@KarthikGanesan88 
Hi! 

TL;DR answer:
Please try to set smaller learning rate ([alpha](https://github.com/tiny-dnn/tiny-dnn/blob/master/examples/mnist/train.cpp#L94) in the example).

Detailed answer:
A learning rate is *very* important parameter for training neural network. Too large learning rate will cause divergence of network parameters, and too small learning rate will cause local minima. Since you got the result as worse as random guess (~10%), the weights of your network might be diverged. So  probably you need to set smaller learning rate.

> The optimal learning rate is usually close to (by a factor of 2) the largest learning rate that does not cause divergence of the training criterion, an observation that can guide heuristics for setting the learning rate.
from http://arxiv.org/pdf/1206.5533v2.pdf ,section 2.1

Another resource to learn how the learning rate is important: (set larger learning rate and see what will occur)
http://playground.tensorflow.org/

**Comment by KarthikGanesan88** [comment]
@nyanp Hi, seems like that was the answer! Initially, the value was set to sqrt(10) = 3.16 and that didnt work. I set it to 2 and that didnt work either! so i went all the way and set alpha to 0.1 and the numbers look ok now! i guess i incorrectly assumed that a learning rate that works for networks with conv -> pool -> fc would work for just fc networks!
